[REFACTOR]: QnA Exception 변경

gooot opened the pull request:
## What is this PR? 🔍

- 기능 : QnA Exception 변경
- issue : #225 

## Changes 📝

SuccessCode 사용하도록 변경

ValidationException 누락 추가

## Precaution

## ✔️ Please check if the PR fulfills these requirements

- [ ] It's submitted to the correct branch, not the `develop` branch unconditionally?
- [ ] If on a hotfix branch, ensure it targets `main`?
- [ ] There are no warning message when you run `yarn lint`

- Close #225

gooot committed:
refactor : QnA Exception 변경

* refactor : QnA Exception 변경

ss0ngcode commented on omocha-domain/src/main/java/org/omocha/domain/qna/answer/AnswerValidatorImpl.java:
```diff
@@ -19,9 +20,9 @@ public class AnswerValidatorImpl implements AnswerValidator {
-		// 	throw new InvalidMemberException(INVALID_MEMBER);
-		// }
+		if (!auction.getMemberId().equals(member.getMemberId())) {
+			throw new MemberInvalidMemberException(member.getMemberId());
```
public class AuctionOwnerMismatchException extends AuctionException {
	public AuctionOwnerMismatchException(Long memberId) {
		super(
			ErrorCode.AUCTION_MEMBER_INVALID,
			"경매 작성자와 일치하지 않는 사용자입니다. memberId: " + memberId
		);
	}
}

이거로 변경해서 사용하시죠!

ss0ngcode commented on omocha-domain/src/main/java/org/omocha/domain/qna/answer/AnswerValidatorImpl.java:
```diff
@@ -19,9 +20,9 @@ public class AnswerValidatorImpl implements AnswerValidator {
-		// 	throw new InvalidMemberException(INVALID_MEMBER);
-		// }
+		if (!auction.getMemberId().equals(member.getMemberId())) {
+			throw new MemberInvalidMemberException(member.getMemberId());
```
```
public class AuctionOwnerMismatchException extends AuctionException {
	public AuctionOwnerMismatchException(Long memberId) {
		super(
			ErrorCode.AUCTION_MEMBER_INVALID,
			"경매 작성자와 일치하지 않는 사용자입니다. memberId: " + memberId
		);
	}
}
```

이걸로 변경해서 사용하시죠!

gooot replied:
변경하였습니다.

gooot committed:
refactor : Auction Exception 변경

* refactor : Auction Exception 변경
- AuctionOwnerMismatchException 추가
- MemberInvalidException 삭제
- MemberInvalidMemberException 삭제

ss0ngcode reviewed (approved):
LGTM!
